magma/simulator/simulator.py

```diff
@@ -27,13 +27,21 @@ def set_value(self, bit, scope, newval):
         pass
 
     @abstractmethod
-    def advance(self, n):
+    def advance_cycle(self, cycles):
+        pass
```

Comment:
Adding this causes existing tests to fail since the PythonSimulator doesn't support these abstractmethods. I can update it to have them since it's simple, but in theory could we define these methods in terms of the abstract `advance` and `rewind` functions?

Reply:
I'll just merge and test this out myself